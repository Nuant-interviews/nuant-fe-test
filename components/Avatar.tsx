export const AvatarSize = {
  large: 160,
  medium: 24,
  small: 40,
} as const;
export function Avatar({
  src,
  alt = "avatar",
  size = "small",
  className,
}: {
  src: string;
  alt: string;
  size?: keyof typeof AvatarSize;
  className?: string;
}) {
  return (
    <img
      style={{ width: AvatarSize[size], height: AvatarSize[size] }}
      className={`z-40 inline-block rounded-full ring-opacity-50 ${className}`}
      src={src}
      alt={alt}
    />
  );
}
