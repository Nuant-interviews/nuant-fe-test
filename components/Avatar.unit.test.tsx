import { expect, test } from "vitest";
import { render } from "@testing-library/react";
import { Avatar } from "./Avatar";

test("Page", () => {
  const { container } = render(
    <Avatar
      src={`https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/home/bulbasaur.png`}
      alt={`Test's Avatar`}
    />
  );

  expect(container).toMatchSnapshot();
});
