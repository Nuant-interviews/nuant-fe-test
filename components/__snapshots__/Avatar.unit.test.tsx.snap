// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Page 1`] = `
<div>
  <img
    alt="Test's Avatar"
    class="z-40 inline-block rounded-full ring-opacity-50 undefined"
    src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/home/bulbasaur.png"
    style="width: 40px; height: 40px;"
  />
</div>
`;
